notary-commons/src/main/kotlin/provider/D3ClientProvider.kt

```diff
@@ -0,0 +1,27 @@
+package provider
+
+import com.github.kittinunf.result.Result
+import com.github.kittinunf.result.map
+import jp.co.soramitsu.iroha.java.QueryAPI
+import model.D3Client
+import sidechain.iroha.util.getAccountDetails
+
+/**
+ * Class that is used to get information about D3 clients
+ */
+class D3ClientProvider(private val notaryQueryAPI: QueryAPI) {
+
+    /**
+     * Returns D3 client by its id
+     * @param accountId - account id of client
+     * @return D3 client
+     */
+    fun getClient(accountId: String): Result<D3Client, Exception> {
+        // Assuming that client sets details himself
```

Comment:
hmm it's needed to be decided
probably `registration_service` could also take care of it

Reply:
Unfortunately, we use different `registration_service` accounts for Bitcoin and Ethereum. I suppose that crypto related details will be set by `registration_service` while common information (such as email) will be set by user himself.